GoogleForCreators/web-stories-wp
# Media Library: add keyboard navigation

## Summary

This PR adds keyboard navigation to the media gallery (both Upload and Media3P tabs) by implementing the "roving tabindex" aria recommendation.
See https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_roving_tabindex

## Relevant Technical Choices

Decided not to use aria-activedescendant as it's not clear at all what the browser support is.

## To-do

I believe there's an issue with keyboard control in the canvas where even though the focus is in the gallery, using the kyeboard results in elements in the canvas moving.

## User-facing changes

When tabbing through the editor and reaching the gallery, the first element in the gallery is outlined. At that point you can use Up/Down/Left/Right/Home/End to navigate the gallery. Tabbing again exits the gallery onto the next tabindex (in the canvas I believe).

If a media element is selected and the user presses Enter, the media element is added to the canvas (equivalent to clicking the element).

## Testing Instructions

Enable the rowBasedGallery and media3pTab flags.

See user-facing changes.

---



Fixes #1305
